File: kotlin/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.kt
class Solution {
    fun numOfSubarrays(arr: IntArray, k: Int, threshold: Int): Int {
        
        var sum = 0
        var res = 0

        for (i in 0 until k-1) sum += arr[i]

        for (i in k-1 until arr.size) {
            sum += arr[i]
            if (sum / k >= threshold) res++
            sum -= arr[i-k+1]
        }

        return res
    }
}

File: go/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.go
func numOfSubarrays(arr []int, k int, threshold int) int {
	res := 0

	var sum func([]int) int
	sum = func(nums []int) int {
		r := 0
		for _, n := range nums {
			r += n
		}
		return r
	}

	curSum := sum(arr[:k-1])

	for i := k - 1; i < len(arr); i++ {
		curSum += arr[i]
		if (curSum / k) >= threshold {
			res += 1
		}
		curSum -= arr[i-k+1]
	}

	return res
}
